packages/rum-core/src/domain/rumEventsCollection/action/getActionNameFromElement.ts

```diff
@@ -175,6 +175,18 @@ function getTextualContent(element: Element | HTMLElement) {
         }
       }
     }
+	
+	const elementsTextToScrub: NodeListOf<HTMLElement> = element.querySelectorAll(`[${DEFAULT_PROGRAMMATIC_ATTRIBUTE}]`)
+
+    // replace the innertext of the elements with programmatic attribute value
+    for (let i = 0; i < elementsTextToScrub.length; i += 1) {
+      const innerText = elementsTextToScrub[i].innerText
+      // eslint-disable-next-line @typescript-eslint/prefer-for-of
+      if (innerText.trim().length > 0) {
+        text = text.replace(innerText, elementsTextToScrub[i].getAttribute(`${DEFAULT_PROGRAMMATIC_ATTRIBUTE}`) || '')
```

Comment:
```suggestion
        text = text.replace(innerText, elementsTextToScrub[i].getAttribute(DEFAULT_PROGRAMMATIC_ATTRIBUTE) || '')
```

Reply:
Addressed.. Taken the suggestion

Reply:
Thank you for reviewing the changes. Please let me know if you have any comments/suggestions